### PR TITLE
Fixed #760

### LINKED ISSUE #760: Relations with query not work
We have this class:

```
<?php

namespace App;

use App\Comment;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Post extends Model
{
    public function comments(): HasMany
    {
        return $this->hasMany(Comment::class);
    }
    
    public function commentsUnread(): HasMany
    {
        return $this->comments()->where('unread', true);
    }
}
```

After run ```php artisan ide-helper:models``` we got: 
```
/**
 * @property-read \Illuminate\Database\Eloquent\Collection|\App\Comment[] $comments
 */
```

But expected:
```
/**
 * @property-read \Illuminate\Database\Eloquent\Collection|\App\Comment[] $comments
 * @property-read \Illuminate\Database\Eloquent\Collection|\App\Comment[] $commentsUnread
 */
```

Seems like because ide-helper just search functions with relations [names](https://github.com/barryvdh/laravel-ide-helper/blob/master/src/Console/ModelsCommand.php#L468) and ignore method return type.

Anyone have workaround for this?

### PR DESCRIPTION
**Pull request by UksusoFF**
Because you [drop support](https://github.com/barryvdh/laravel-ide-helper/commit/09db8c9a76711e98c61af0795934fb15955223fb#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780) php < 7 we can use [ReflectionFunctionAbstract::getReturnType](http://php.net/manual/en/reflectionfunctionabstract.getreturntype.php).

This PR tested for relations with return type and without.

### CODE REVIEW

File: src/Console/ModelsCommand.php [hunk 1]
```diff
@@ -450,81 +450,73 @@ protected function getPropertiesFromMethods($model)
...
+                    }
 
-                    $code = '';
-                    while ($file->key() < $reflection->getEndLine()) {
-                        $code .= $file->current();
-                        $file->next();
+                    $returnType = $reflection->getReturnType();
+
+                    if ($returnType === null) {
+                        $relationObj = $model->$method();
```
Comment (edcoreweb) [inline src/Console/ModelsCommand.php]: You might trigger some unwanted behaviour here. You are basically calling every method on the model. (Imagine you have a method which sends a mail or something.)

**Comment by edcoreweb** [comment]
> Yeah perhaps use the return type if possible, otherwise old behavior.

I did a minimal implementation of this issue as well: https://github.com/barryvdh/laravel-ide-helper/pull/763

**Comment by barryvdh** [comment]
Merged the other one, thanks!

**Comment by UksusoFF** [comment]
> You might trigger some unwanted behaviour here. You are basically calling every method on the model. (Imagine you have a method which sends a mail or something.)
@edcoreweb I don't really worried about this on development environment (laravel-ide-helper should be used only on dev right?).
Seems like PR #763 maybe more safety for any other.

**Pull request closed without merging**